src/startupscriptgenerator/node/scriptgenerator.go

```diff
@@ -215,6 +241,56 @@ func (gen *NodeStartupScriptGenerator) getPackageJsonMainCommand(packageJsonObj
 	return ""
 }
 
+func (gen *NodeStartupScriptGenerator) getProcessJsonCommand(userInputPath string) string {
+	processJsonPath := ""
+	if userInputPath != "" {
+		if strings.HasSuffix(userInputPath, ".json") {
+			processJsonPath = userInputPath
+		}
+	} else {
+		processJsonPath = "process.json"
+	}
+
+	if processJsonPath != "" {
+		processJsonFullPath := filepath.Join(gen.SourcePath, processJsonPath)
+		if common.FileExists(processJsonFullPath) {
+			debugCommand := ""
+			if gen.isDebugging() {
+				debugCommand = gen.getNodeWrapperDebugCommand()
+			}
+			return debugCommand + getPm2StartCommand(processJsonPath)
+		}
+	}
+	return ""
+}
+
+func (gen *NodeStartupScriptGenerator) getConfigYamlCommand(userInputFullPath string) string {
```

Comment:
This method looks different than others where you full path etc....intentional?

Reply:
I think these three methods could be consolidated into one and parameterized..what do you think?

Reply:
ok, refactored it a bit. Let me know what you think.